Functions/Admin/Set-RsDatabase.ps1

```diff
@@ -1,172 +1,241 @@
+            Set-RsDatabase -DatabaseServerName localhost -Name ExistingReportServer -IsExistingDatabase -Authentication 0 -DatabaseCredential $myCredentials
             Description
-            ----------- 
+            -----------
             This command will configure Reporting Services to connect to an existing RS database (ExistingReportServer) using Windows credentials ($myCredentials).
+        
+        .NOTES
+            Author:      ???
+            Editors:     Friedrich Weinmann
+            Created on:  ???
+            Last Change: 31.01.2017
+            Version:     1.1
+            
+            Release 1.1 (26.01.2017, Friedrich Weinmann)
+            - Fixed Parameter help (Don't poison the name with "(optional)", breaks Get-Help)
+            - Standardized the parameters governing the Report Server connection for consistent user experience.
+            - Renamed the parameter 'DatabaseCredentialType' to 'Authentication', in order to maintain parameter naming conventions. Added the previous name as an alias, for backwards compatiblity.
+            - Replaced the parametertype of -Authentication to the new enumeration ReportingServicesTools.SqlServerAuthentication.
+            - Removed the parameter validation on Authentication: ValidateSet serves no purpose with an enumerated type.
+            - Renamed the parameter 'DatabaseName' to 'Name', in order to maintain parameter naming conventions. Added the previous name as an alias, for backwards compatiblity.
+            - Altered the way "New-RsConfigurationSettingObject" is called to the default template. This offers a consistent experience across all WMI-based functions.
+            - Replaced calling exit with throwing a terminating error (exit is a bit of an overkill when failing a simple execution)
+            - Improved error message on failure.
+            - Implemented ShouldProcess (-WhatIf, -Confirm)
+            
+            Release 1.0 (???, ???)
+            - Initial Release
     #>
 
-    [cmdletbinding()]
-    param(
-        [string]
-        $ReportServerInstance='MSSQLSERVER',
-
-        [string]
-        $ReportServerVersion ='13',
-
-        [Parameter(Mandatory=$True)]
+    [CmdletBinding(SupportsShouldProcess = $true, ConfirmImpact = 'High')]
+    param (
+        [Parameter(Mandatory = $True)]
         [string]
         $DatabaseServerName,
 
         [switch]
         $IsRemoteDatabaseServer,
-
-        [Parameter(Mandatory=$True)]
+        
+        [Parameter(Mandatory = $True)]
+        [Alias('DatabaseName')]
         [string]
-        $DatabaseName,
+        $Name,
 
         [switch]
         $IsExistingDatabase,
-
-        [Parameter(Mandatory=$True)]
-        [ValidateSet('Windows','SQL','ServiceAccount')]
+        
+        [Parameter(Mandatory = $true)]
+        [Alias('DatabaseCredentialType')]
+        [ReportingServicesTools.SqlServerAuthentication]
```

Comment:
Will do. How about the namespace in general? Make it 'Microsoft.ReportingServicesTools.SqlServerAuthenticationType'? Or maybe even 'Microsoft.SqlServer.ReportingServicesTools.SqlServerAuthenticationType'?